src/v/kafka/server/handlers/describe_configs.cc

```diff
@@ -334,7 +335,11 @@ static void report_broker_config(
       config::shard_local_cfg().delete_retention_ms,
       include_synonyms,
       [](const std::chrono::milliseconds& ms) {
-          return ssx::sformat("{}", ms.count());
+          if (ms == std::chrono::milliseconds::max()) {
```

Comment:
The problem is that `std::chrono::milliseconds::max()` is a perfectly valid value to enter into the configuration.

One solution is to validate the property as only allowing up to `max()-1`, but the hack continues.

Reply:
We have `tristate` class in source tree, maybe using that would help ?
https://github.com/redpanda-data/redpanda/blob/dev/src/v/tristate.h